ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/Performance.java

```diff
@@ -0,0 +1,237 @@
+        }
+
+        sizeOfFiles = args.getSizeOfFiles();
+        if (sizeOfFiles == null) {
+            throw new MissingOptionException("The performance command requires '-s' to be set.");
+        }
+
+        return this;
+    }
+
+    @Override
+    public PerformanceResult call() throws Exception {
+        final Ds3ClientHelpers helpers = Ds3ClientHelpers.wrap(getClient());
+        final int numberOfFiles = Integer.valueOf(this.numberOfFiles);
+        final long sizeOfFiles = Integer.valueOf(this.sizeOfFiles);
+
+        try {
+            helpers.ensureBucketExists(this.bucketName);
+
+            final List<Ds3Object> objList = Lists.newArrayList();
+            for (int i = 0; i<numberOfFiles ; i++) {
+                final long testFileSize = sizeOfFiles * 1024L * 1024L;
+                final Ds3Object obj = new Ds3Object("file_" + i, testFileSize);
+                objList.add(obj);
+            }
+
+            /**** PUT ****/
+            final Ds3ClientHelpers.Job putJob = helpers.startWriteJob(this.bucketName, objList);
+            final PerformanceListener putPerformanceListener = new PerformanceListener(
+                    System.currentTimeMillis(),
+                    numberOfFiles,
+                    numberOfFiles * sizeOfFiles,
+                    true);
+            putJob.attachObjectCompletedListener(putPerformanceListener);
+            putJob.attachDataTransferredListener(putPerformanceListener);
+            putJob.transfer(new MemoryObjectChannelBuilder(sizeOfFiles));
+
+            System.out.println();
+
+            /**** GET ****/
+            final Ds3ClientHelpers.Job getJob = helpers.startReadJob(this.bucketName, objList);
+            final PerformanceListener getPerformanceListener = new PerformanceListener(
+                    System.currentTimeMillis(),
+                    numberOfFiles,
+                    numberOfFiles * sizeOfFiles,
+                    false);
+            getJob.attachObjectCompletedListener(getPerformanceListener);
+            getJob.attachDataTransferredListener(getPerformanceListener);
+            getJob.transfer(new MemoryObjectChannelBuilder(sizeOfFiles));
+
+        } finally {
+            deleteAllContents(getClient(), this.bucketName);
+        }
+        return new PerformanceResult("");
+    }
+
+    private void deleteAllContents(final Ds3Client client, final String bucketName) throws IOException, SignatureException {
+        final Ds3ClientHelpers helpers = Ds3ClientHelpers.wrap(client);
+
+        final Iterable<Contents> objects = helpers.listObjects(bucketName);
+        for(final Contents contents : objects) {
+            client.deleteObject(new DeleteObjectRequest(bucketName, contents.getKey()));
+        }
+
+        client.deleteBucket(new DeleteBucketRequest(bucketName));
+    }
+}
+
+class MemoryObjectChannelBuilder implements Ds3ClientHelpers.ObjectChannelBuilder {
```

Comment:
This is not how you typically do inner classes like this in Java.  They should actually be enclosed inside of the Performance class and then made private.